### PR TITLE
[CORE24-192] task(models): create models lib

### PR DESCRIPTION
**Pull request by internetti**
Jira ticket: JIRA-CORE24-192

## What have you implemented?

generated a lib in the nx project for the models, moved User model from api to the lib

## How to test it?

build api

### CODE REVIEW

File: apps/core-api/src/controllers/scim.controller.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import { Router, json } from 'express';
 import { z } from 'zod';
 
+import { User } from '@nrcno/models';
```
Comment (neb42) [inline apps/core-api/src/controllers/scim.controller.ts:4]: `@nrcno/core-models`

File: libs/models/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "@nrcno/models",
+  "version": "0.0.1",
+  "dependencies": {
+    "tslib": "^2.3.0",
+    "zod": "^3.22.4"
+  },
+  "type": "commonjs",
+  "main": "./src/index.js",
+  "typings": "./src/index.d.ts",
+  "private": true
```
Comment (neb42) [inline libs/models/package.json:11]: Unsure why nx adds all this. I don't think it's needed.

File: libs/models/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@nrcno/models",
```
Comment (neb42) [inline libs/models/package.json:2]: `@nrcno/core-models`

File: libs/models/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "extends": "../../tsconfig.base.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "noImplicitOverride": true,
+    "noPropertyAccessFromIndexSignature": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true
```
Comment (neb42) [inline libs/models/tsconfig.json:10]: All of this is in the base tsconfig

File: tsconfig.base.json [hunk 5]
```diff
@@ -10,7 +10,8 @@
     "module": "esnext",
     "moduleResolution": "node",
     "paths": {
-      "@nrcno/db": ["libs/db/src/index.ts"]
+      "@nrcno/db": ["libs/db/src/index.ts"],
+      "@nrcno/models": ["libs/models/src/index.ts"]
```
Comment (neb42) [inline tsconfig.base.json:14]: `@nrcno/core-models`

**Review by neb42 — CHANGES_REQUESTED** [review]

File: apps/core-api/src/controllers/scim.controller.ts (hunk 1 above)
Comment (internetti) [reply apps/core-api/src/controllers/scim.controller.ts:4]: ah, misunderstanding, I thought we wanted to drop the core- alltogether for the lib

File: libs/models/tsconfig.json (hunk 4 above)
Comment (internetti) [reply libs/models/tsconfig.json:10]: the only thing I see duplicated in the base config is strict: true?

Comment (neb42) [reply libs/models/tsconfig.json]: ah I've added the others in another pr and I'm not a fan of `noPropertyAccessFromIndexSignature`

**Review by neb42 — APPROVED** [review]

**Pull request merged**